tests/Scout/ScoutIntegrationTest.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+namespace MongoDB\Laravel\Tests\Scout;
+
+use DateTimeImmutable;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\LazyCollection;
+use Laravel\Scout\ScoutServiceProvider;
+use LogicException;
+use MongoDB\Laravel\Tests\Scout\Models\ScoutUser;
+use MongoDB\Laravel\Tests\Scout\Models\SearchableInSameNamespace;
+use MongoDB\Laravel\Tests\TestCase;
+use Override;
+use PHPUnit\Framework\Attributes\Depends;
+
+use function array_merge;
+use function count;
+use function env;
+use function Orchestra\Testbench\artisan;
+use function range;
+use function sprintf;
+use function usleep;
+
+class ScoutIntegrationTest extends TestCase
+{
+    #[Override]
+    protected function getPackageProviders($app): array
+    {
+        return array_merge(parent::getPackageProviders($app), [ScoutServiceProvider::class]);
+    }
+
+    #[Override]
+    protected function getEnvironmentSetUp($app): void
+    {
+        parent::getEnvironmentSetUp($app);
+
+        $app['config']->set('scout.driver', 'mongodb');
+        $app['config']->set('scout.prefix', 'prefix_');
+    }
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        // Init the SQL database with some objects that will be indexed
+        // Test data copied from Laravel Scout tests
+        // https://github.com/laravel/scout/blob/10.x/tests/Integration/SearchableTests.php
+        ScoutUser::executeSchema();
+
+        $collect = LazyCollection::make(function () {
+            yield ['name' => 'Laravel Framework'];
+
+            foreach (range(2, 10) as $key) {
+                yield ['name' => 'Example ' . $key];
+            }
+
+            yield ['name' => 'Larry Casper', 'email_verified_at' => null];
+            yield ['name' => 'Reta Larkin'];
+
+            foreach (range(13, 19) as $key) {
+                yield ['name' => 'Example ' . $key];
+            }
+
+            yield ['name' => 'Prof. Larry Prosacco DVM', 'email_verified_at' => null];
+
+            foreach (range(21, 38) as $key) {
+                yield ['name' => 'Example ' . $key, 'email_verified_at' => null];
+            }
+
+            yield ['name' => 'Linkwood Larkin', 'email_verified_at' => null];
+            yield ['name' => 'Otis Larson MD'];
+            yield ['name' => 'Gudrun Larkin'];
+            yield ['name' => 'Dax Larkin'];
+            yield ['name' => 'Dana Larson Sr.'];
+            yield ['name' => 'Amos Larson Sr.'];
+        });
+
+        $id = 0;
+        $date = new DateTimeImmutable('2021-01-01 00:00:00');
+        foreach ($collect as $data) {
+            $data = array_merge(['id' => ++$id, 'email_verified_at' => $date], $data);
+            ScoutUser::create($data)->save();
+        }
+
+        self::assertSame(44, ScoutUser::count());
+    }
+
+    /** This test create the search index for tests performing search */
+    public function testItCanCreateTheCollection()
+    {
+        $this->skipIfSearchIndexManagementIsNotSupported();
```

Comment:
Would it make more sense to check for this when the test itself is setup? I assume none of the integration tests can be run without support for search indexes.